### PR TITLE
[rsmq-promise-native] Create type definitions

### PR DESCRIPTION
**Pull request by dubblen**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`

### CODE REVIEW

File: types/rsmq-promise-native/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        "strictFunctionTypes": true,
+        "strictNullChecks": true,
+        "baseUrl": "../",
+        "typeRoots": [
+            "../"
+        ],
+        "types": [],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true,
+        "esModuleInterop": true
```
Comment (amcasey) [inline types/rsmq-promise-native/tsconfig.json:18]: Why was `esModuleInterop` required?

File: types/rsmq-promise-native/package.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+    "private": true,
+    "dependencies": {
+        "@types/rsmq": "^0.8.4"
```
Comment (amcasey) [inline types/rsmq-promise-native/package.json:4]: This shouldn't be necessary for a package in this repo.

**Review by amcasey — CHANGES_REQUESTED** [review]
Please add at least simple tests.

File: types/rsmq-promise-native/rsmq-promise-native-tests.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import RedisSMQPromise from "rsmq-promise-native";
+
+const rsmq = new RedisSMQPromise({host: "127.0.0.1", port: 7001, ns: "rsmq"});
+
+rsmq.createQueue({qname: 'myqueue'})
+    .then(done => {
+    	console.log('Queue created!');
```
Comment (danger-public) [inline types/rsmq-promise-native/rsmq-promise-native-tests.ts:7]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

**Review by amcasey — CHANGES_REQUESTED** [review]
I might be missing something, but my previous comments don't seem to have been addressed.

**Comment by dubblen** [comment]
@amcasey Hi. I have added some simple tests as you asked for. It won't pass the tdtests when I remove the dependencies and tsconfig.json properties that you told me to remove.

**Comment by amcasey** [comment]
@dubblen This is a slightly strange case where `rsmq` has been removed from DefinitelyTyped because the package started publishing its own type.  So, you _do_ need a package.json file but, unless you specifically want a version older than 0.8.4 (which does not appear to be the case), you should depend on `rsmq`, rather than `@types/rsmq`.

**Comment by amcasey** [comment]
Also, there appear to be some stray tab characters in your test file - please clean those up.

**Pull request closed without merging**